src/librbd/api/Image.h

```diff
@@ -59,6 +59,11 @@ struct Image {
                        ProgressContext &prog_ctx);
   static int deep_copy(ImageCtxT *src, ImageCtxT *dest, bool flatten,
                        ProgressContext &prog_ctx);
+  static int deep_copy(ImageCtxT *src, ImageCtxT *dest, 
+                       librados::snap_t src_snap_id_start,
+                       librados::snap_t src_snap_id_end,
+                       librados::snap_t dst_snap_id_start, ImageOptions& opts,
+                      ProgressContext &prog_ctx);
```

Comment:
nit: I am not sure we want this here. Currently it is going to be used only for tests, so may be move this function to the test file? Otherwise it might be confusing: in api directory we have functions that are used for C/C++ API implementation, while this is not used. Actually I would be fine if you added this to API, and then having it here would make sense. If you didn't have plans to add this to API then I would not have it here.

Reply:
Based on a conversation with @idryomov , I have opened a new PR https://github.com/ceph/ceph/pull/52358 for the blkdiscard issue only. 

The TestDeepCopy.CloneIncremental failure is unrelated to the fix and is present in the existing code. I will work on that separately.